src/facade/dragonfly_connection.cc

```diff
@@ -1291,11 +1282,15 @@ void Connection::ClearPipelinedMessages() {
   // As well as to avoid pubsub backpressure leakege.
   for (auto& msg : dispatch_q_) {
     FiberAtomicGuard guard;  // don't suspend when concluding to avoid getting new messages
-    if (msg.IsIntrusive())
+    if (msg.IsControl())
       visit(dispatch_op, msg.handle);  // to not miss checkpoints
     RecycleMessage(std::move(msg));
   }
 
+  if (dispatch_q_.size() == stats_->dispatch_queue_max_len) {
+    // conservative reset that will be corrected upon next insertion.
+    stats_->dispatch_queue_max_len = 0;
```

Comment:
why do you call it dispatch_queue_**max_len** if you update it to be the size of dispatch_queue_?
why not dispatch_queue_len_?